### PR TITLE
[5.8] Clean up cache integration tests

### PR DESCRIPTION
**Pull request by franzliedke**
I noticed the Memcached tests were failing on my machine, because Memcached was not set up properly. The PHP extension was not installed, but could not connect to any server...

I would have expected these to be skipped instead of failing in this scenario.

Other small changes while I was at it:
- Check for Memcached extension to be installed instead of a class to be available (which could, in theory, happen for another reason).
- Split up Memcached and Redis tests, so the latter won't be skipped because the former isn't available.
- For the Redis cache lock tests, use PhpUnit's `setUp` and `tearDown` instead of a custom mechanism.

### CODE REVIEW

**Comment by TBlindaruk** [comment]
@franzliedke , maybe will be better target this PR to 5.7 ?

**Comment by franzliedke** [comment]
Hmm, probably true. Will do that if I get an approval.

File: tests/Integration/Cache/MemcachedCacheLockTest.php [hunk 1]
```diff
@@ -6,22 +6,33 @@
 use Illuminate\Support\Carbon;
 use Orchestra\Testbench\TestCase;
 use Illuminate\Support\Facades\Cache;
-use Illuminate\Foundation\Testing\Concerns\InteractsWithRedis;
 
 /**
  * @group integration
  */
-class CacheLockTest extends TestCase
+class MemcachedCacheLockTest extends TestCase
```
Comment (TBlindaruk) [inline tests/Integration/Cache/MemcachedCacheLockTest.php]: @franzliedke , maybe will be better target this PR to 5.7 ?

**Comment by deleugpn** [comment]
I also see no reason not to target 5.7

**Comment by taylorotwell** [comment]
Yes I think this can be sent to 5.7.

**Pull request closed without merging**